app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -0,0 +1,48 @@
+module Spree::CheckoutControllerDecorator
+  def self.prepended(base)
+    base.before_action :checkout_afterpay_payment, only: [:update]
+  end
+
+  # Afterpay Specific Methods Start
+  def checkout_afterpay_payment
+    if afterpay_payment_present?
```

Comment:
we should not do this in controller here

Reply:
Look into Spree::Gateway and look into how we can use that library

you can check better_spree_paypal for reference